torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -2651,6 +2681,18 @@ def shard_full_optim_state_dict(
             argument to the optimizer, then you should pass that same value to
             this method as ``optim_input``.
 
+        .. note:: Both :meth:`shard_full_optim_state_dict` and
+            :meth:`scatter_full_optim_state_dict` may be used to get the
+            sharded optimizer state dict to load. Assuming that the full
+            optimizer state dict resides in CPU memory, the former requires
+            each rank to have the full dict in CPU memory, where each rank
+            individually shards the dict without any communication, while the
+            latter requires only rank 0 to have the full dict in CPU memory,
+            where rank 0 moves each shard to GPU memory (for NCCL) and
+            communicates it to ranks appropriately. Hence, the former has
+            higher aggregate CPU memory cost, while the latter has higher
+            communication cost.
+
```

Comment:
Added a note above to explain difference between the two APIs. (replicated on both methods)